app/controllers/administrative-units/administrative-unit/legal-structures/new.js

```diff
@@ -0,0 +1,63 @@
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { dropTask } from 'ember-concurrency';
+import { combineFullAddress } from 'frontend-contact-hub/models/address';
+
+export default class AdministrativeUnitsAdministrativeUnitLegalStructuresNewController extends Controller {
+  @service router;
+  @service store;
+
+  @dropTask
+  *createAssociatedStructureTask(event) {
+    event.preventDefault();
+
+    let {
+      associatedStructure,
+      address,
+      legalType,
+      registration,
+      structuredIdentifier,
+    } = this.model;
+
+    yield associatedStructure.validate();
+    yield legalType.validate();
+
+    if (associatedStructure.isValid && legalType.isValid) {
+      address.fullAddress = combineFullAddress(address);
+      yield address.save();
+
+      yield legalType.save();
+
+      yield structuredIdentifier.save();
+      registration.structuredIdentifier = structuredIdentifier;
+      yield registration.save();
+
+      associatedStructure.address = address;
+      associatedStructure.legalType = legalType;
+      associatedStructure.registration = registration;
+      yield associatedStructure.save();
+
+      this.model.administrativeUnit.associatedStructures.pushObject(
```

Comment:
I would destructure `administrativeUnit` in the same way as the other model properties on line 14-20.